### PR TITLE
Show notice on save in site editor

### LINKED ISSUE #36166: Site editor has no visual or screenreader notifications about the result of saving
### Description

When saving in the site editor, there's no visual notification (snackbar message) or screenreader message to indicate saving has completed successfully or unsuccessfully (like when offline). I think this needs to be solved before shipping the editor.

A snackbar (on success) or error notice (on failure) should be quick wins that would solve this.

A second part that could be improved is the entity saving panel. When clicking the first 'Save' button, a second save button is focused and testing through Voiceover I just head 'Button, Save, publish' again, which sounds buggy, almost like the first button didn't do anything. It could be an improvement to add a description like 'Confirm your changes' to this button. At least something that indicates there is more content below the second save button.

### Step-by-step reproduction instructions

With a screenreader active:
1. Open the site editor
2. Make some changes to the content
3. Focus the save button and hit enter.
4. Hear that a save button has been focused again, but what actually happened after clicking the first button is a bit hard to follow
5. Optionally set network throttling offline in the browser dev tools
6. Press save again
7. Observe there's no message indicating whether saving was successful or unsuccessful

### Screenshots, screen recording, code snippet

_No response_

### Environment info

Tested in Voiceover / Safari
Tested network throttling in Brave, since Safari doesn't have this feature 😞 

### Please confirm that you have searched existing issues in the repo.

Yes

### Please confirm that you have tested with all plugins deactivated except Gutenberg.

Yes

### PR DESCRIPTION
**Pull request by tellthemachines**
## Description

Fixes #36166. Adds notices on success and failure to the Save action in the site editor.

## How has this been tested?

In the site editor, make changes to one or two template parts and click save. Check that a snackbar pops up when save is successful. 
Make some more changes and set network to "offline". Saving again should cause an error notice to appear.

Repeat this process using a screen reader, and check that the notices are read by the screen reader when they pop up.

I tested this with VoiceOver in Safari, Firefox and Chrome. Safari and Chrome announce the notices correctly, but Firefox only announces the failure notice. It seems to ignore the snackbar altogether.

## Types of changes

Bug fix (non-breaking change which fixes an issue)

## Checklist:
- [ ] My code is tested.
- [ ] My code follows the WordPress code style. 
- [ ] My code follows the accessibility standards. 
- [ ] I've tested my changes with keyboard and screen readers. 
- [ ] My code has proper inline documentation. 
- [ ] I've included developer documentation if appropriate. 
- [ ] I've updated all React Native files affected by any refactorings/renamings in this PR (please manually search all `*.native.js` files for terms that need renaming or removal).

### CODE REVIEW

**Review by getsource — COMMENTED** [review]
Thank you! This looks great.

I tested, and it works as expected.

In the below videos, I tested first by clicking the Save button twice, then by using CMD+S, followed by the return key, pressing the tab key after to see if the next focus targets would be the same. The next target for tab was the same both before and after applying the PR, as expected.

Before:

https://user-images.githubusercontent.com/1034160/143547603-08a5a42e-3aed-4ef4-a0a4-ce6b9433a89f.mov

After:

https://user-images.githubusercontent.com/1034160/143547638-de6fb707-abc7-4cc5-bfdf-c6ba3ba51e6f.mov

File: packages/editor/src/components/entities-saved-states/index.js [hunk 1]
```diff
@@ -148,17 +154,31 @@ export default function EntitiesSavedStates( { close } ) {
...
+				saveSpecifiedEntityEdits(
+					'root',
+					'site',
+					undefined,
+					siteItemsToSave
+				)
 			);
 		}
+
+		Promise.all( pendingSavedRecords ).then( ( values ) => {
```
Comment (kevin940726) [inline packages/editor/src/components/entities-saved-states/index.js]: Could `pendingSavedRecords` fail? Might be easy enough to also add `catch` to handle it with error notices too.

Comment (tellthemachines) [reply packages/editor/src/components/entities-saved-states/index.js]: `pendingSavedRecords` is an array of the results of calling `saveEntityRecord`, which afaik never returns an error, just a record or `undefined`. That's why I'm checking for 'undefined' when showing the failure notice. Would it still be worth adding a `catch` in this case? We wouldn't really be doing anything with the caught error.

Comment (kevin940726) [reply packages/editor/src/components/entities-saved-states/index.js]: That's a bit unexpected TBH, why wouldn't `saveEntityRecord` throw an error if it... errors? I guess adding a `catch` case does no harm either, in case we want to revisit the behavior of `saveEntityRecord`.

Comment (tellthemachines) [reply packages/editor/src/components/entities-saved-states/index.js]: I'm not sure what the history behind this is, but the savePost function accounts for it by fetching errors in a [super roundabout way](https://github.com/WordPress/gutenberg/blob/trunk/packages/editor/src/store/actions.js#L229). I'm guessing any changes to `saveEntityRecord` at this point would require a fair bit of refactoring anyway 😅 

Not to say we can't add in a `catch` here, it won't hurt to do so. But we'll still have to generate the failure notices based on `undefined` as the return value too.

**Comment by talldan** [comment]
I merged a fix for the unit test issue - https://github.com/WordPress/gutenberg/pull/36987

**Comment by alexstine** [comment]
@tellthemachines The notice is announced for me.

It says "Site updated."

Looks like it uses @wordpress/a11y.speak(). If some screen readers don't announce, you may need to force "assertive" as second parameter to that function. However, it seems to use this in the module you imported I guess, not directly. createSuccessNotice.

**Comment by carolinan** [comment]
Tested with NVDA on Windows in Chrome and Firefox. The successful update and the failure are announced in both browsers.

**Review by kevin940726 — APPROVED** [review]
Oh hey I also pushed some commits here so I obviously approve it 😆 . I'm not cheating.
